srml/staking/src/phragmen.rs

```diff
@@ -207,162 +212,180 @@ pub fn elect<T: Trait + 'static, FV, FN, FS>(
-				if let Some(c) = elected_candidates.iter_mut().find(|c| c.who == e.who) {
-					e.elected = true;
-					e.backing_stake = n.budget.saturating_mul(*e.load) / n.load.max(1);
-					c.backing_stake = c.backing_stake.saturating_add(e.backing_stake);
-					if c.who != n.who {
-						// Only update the exposure if this vote is from some other account.
-						c.exposure.total = c.exposure.total.saturating_add(into_votes(e.backing_stake));
-						c.exposure.others.push(
-							IndividualExposure { who: n.who.clone(), value: into_votes(e.backing_stake) }
-						);
+				if let Some(c) = elected_candidates.iter().find(|c| **c == e.who) {
+					if *c != n.who {
+						let ratio = {
+							// Full support. No need to calculate.
+							if *n.load == *e.load { ACCURACY }
+							else {
+								// This should not saturate. Safest is to just check
+								if let Some(r) = ACCURACY.checked_mul(*e.load) {
+									r / n.load.max(1)
+								} else {
+									// Just a simple trick.
+									*e.load / (n.load.max(1) / ACCURACY)
+								}
+							}
+						};
+						e.ratio = ratio;
+						assignment.1.push((e.who.clone(), ratio));
 					}
 				}
 			}
-		}
-
-		// Optionally perform equalize post-processing.
-		if config.equalize {
-			let tolerance = config.tolerance;
-			let equalize_iterations = config.iterations;
 
-			// Fix indexes
-			nominators.iter_mut().for_each(|n| {
-				n.edges.iter_mut().for_each(|e| {
-					if let Some(idx) = elected_candidates.iter().position(|c| c.who == e.who) {
-						e.elected_idx = idx;
-					}
-				});
-			});
-
-			for _i in 0..equalize_iterations {
-				let mut max_diff = <BalanceOf<T>>::zero();
-				nominators.iter_mut().for_each(|mut n| {
-					let diff = equalize::<T>(&mut n, &mut elected_candidates, tolerance);
-					if diff > max_diff {
-						max_diff = diff;
-					}
-				});
-				if max_diff < tolerance {
-					break;
+			if assignment.1.len() > 0 {
+				// To ensure an assertion indicating: no stake from the nominator going to waste,
+				// we add a minimal post-processing to equally assign all of the leftover stake ratios.
+				let sum = assignment.1.iter().map(|a| a.1).sum::<ExtendedBalance>();
+				let diff = ACCURACY.checked_sub(sum).unwrap_or(0) as usize;
+				let l = assignment.1.len();
+				// TODO: rethink this approach + the limit value. There should indeed be a limit.
+				for i in 0..diff.min(1000) {
+					assignment.1[i%l].1 = assignment.1[i%l].1.saturating_add(1);
 				}
+				assigned.push(assignment);
 			}
 		}
+
 	} else {
 		// if we have less than minimum, use the previous validator set.
 		return None
 	}
-	Some(elected_candidates)
+	Some((elected_candidates, assigned))
 }
 
 /// Performs equalize post-processing to the output of the election algorithm
 /// This function mutates the input parameters, most noticeably it updates the exposure of
 /// the elected candidates.
+///
 /// The return value is to tolerance at which the function has stopped.
```

Comment:
There is no more return value. Does this break API?

Reply:
Good catch to update the doc but nothing breaking; these are just internal functions.